### PR TITLE
feat: fallback to latest supported patch version

### PR DESCRIPTION
**Pull request by vadson71**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -117,6 +117,25 @@
...
     },
+    {
+      "type": "node",
+      "request": "launch",
+      "name": "XML completion: Run Current Jest File",
+      "program": "${workspaceFolder}/node_modules/.bin/jest",
+      "args": [
+        "${fileBasenameNoExtension}",
+        "--coverage=false",
+        "--config",
```
Comment (voicis) [inline .vscode/launch.json:128]: We probably should start the tests only with one worker here as well. (pass `--maxWorkers=1`)

Comment (vadson71) [reply .vscode/launch.json:133]: done

**Review by voicis — APPROVED** [review]

File: packages/context/src/ui5-model.ts [hunk 2]
```diff
@@ -362,141 +466,319 @@ export async function negotiateVersionWithFetcher(
...
+): Promise<{
+  resolvedVersion: string;
+  isIncorrectVersion: boolean;
+  remember: boolean;
+}> {
+  let resolvedVersion = requestedVersion;
+  let isIncorrectVersion = false;
+  if (
+    !useLatestVersion &&
+    (await isVersionResourcesAvailable(
```
Comment (marufrasully) [inline packages/context/src/ui5-model.ts:679]: will that restricted check not cause this function to return false due to the fact that there is empty object saved in cache?

File: packages/context/test/unit/ui5-model.test.ts [hunk 3]
```diff
@@ -11,20 +32,72 @@ import {
...
+  VersionMapJsonType,
 } from "../../src/ui5-model";
 import { FetchResponse } from "@ui5-language-assistant/language-server";
+import { fetch } from "@ui5-language-assistant/logic-utils";
+import * as logicUtils from "@ui5-language-assistant/logic-utils";
+import { getVersionJsonUrl } from "../../src/utils";
+import semverMinSatisfying from "semver/ranges/min-satisfying";
+import { Response } from "node-fetch";
+
+const GET_MODEL_TIMEOUT = 30000;
```
Comment (marufrasully) [inline packages/context/test/unit/ui5-model.test.ts:44]: any specific reason to have this timeout? By default it pick this https://github.com/SAP/ui5-language-assistant/blob/master/jest.setup.js#L1

File: packages/context/src/ui5-model.ts [hunk 4]
```diff
@@ -279,14 +292,92 @@ async function getVersionInfo(
...
+      let apiJson: Json | undefined;
+      try {
+        apiJson = await readFromCache(cacheFilePath);
+      } catch (e) {
+        apiJson = undefined;
+      }
+      return (
+        apiJson &&
+        typeof apiJson === "object" &&
+        Object.keys(apiJson).length > 0
```
[the remark this answers is not in the document]
Comment (vadson71) [reply packages/context/src/ui5-model.ts:323]: Removed last condition of the logic expression

File: packages/context/src/ui5-model.ts [hunk 5]
```diff
@@ -279,14 +292,92 @@ async function getVersionInfo(
...
+
+/**
+ * Checks whether the test resources for specific version are available locally or on server.
+ *
+ * @param versionInfoJsonFetcher Fetcher function which loads version info json
+ * @param versionLibsFetcher Fetcher function which loads version libs
+ * @param modelCachePath Path to model data cache on disk
+ * @param framework The UI5 framework version
+ * @param version Requested version
+ * @returns true if resouces are found
```
[the remark this answers is not in the document]
Comment (vadson71) [reply packages/context/src/ui5-model.ts:338]: fixed

File: packages/context/src/ui5-model.ts (hunk 2 above)
Comment (vadson71) [reply packages/context/src/ui5-model.ts:679]: fixed by above mentioned cache content check condition change

**Pull request merged**